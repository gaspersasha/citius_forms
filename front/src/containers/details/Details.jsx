import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { Spinner, Button, Typography } from '~components';
import { useDetailsContext } from '~contexts';
import { INPUT_STATUS, VALIDATION_TYPES } from '~constants';
import { validate, scrollUpTo, getFormFieldsNames } from '~utils';
import { Done, Exclamation } from '~assets/svg';
import { About, Additional, Address, Contact, SignIn } from '.';
import { composeDetailsState } from './helpers';
import s from './styles/details.module.sass';

const Details = () => {
  const { state, actions } = useDetailsContext();
  const { isLoadingGet, isLoadingPost, updated, apiError, data } = state;
  const [fields, setFieldsData] = useState(composeDetailsState(data));
  const names = getFormFieldsNames(fields);

  useEffect(actions.getDetails, []);

  useEffect(() => {
    setFieldsData(composeDetailsState(data));
  }, [data]);

  useEffect(() => {
    if (apiError || updated) {
      scrollUpTo('#root');
    }
  }, [apiError, updated]);

  const handleChange = ({ target: { name, value, checked } }) => {
    const field = { value };

    if (updated || apiError) {
      actions.setDetailsState({ apiError: false, updated: false });
    }

    switch (name) {
      case names.validUKPassport:
        field.value = value === 'Yes';
        break;

      case names.subscribed:
        field.value = !checked;
        break;

      case names.dependants:
        if (value < 0) {
          return;
        }

        field.value = Number.isNaN(parseInt(value, 10)) ? '' : value;
        break;

      case names.phone:
      case names.alternativePhone:
        if (value !== '0' && value.length && !Number(value)) {
          return;
        }

        break;

      case names.postcode:
        if (typeof value !== 'string') {
          const fieldsToUpdate = {};

          Object.keys(value).forEach((key) => {
            fieldsToUpdate[key] = {
              ...fields[key],
              value: value[key],
            };
          });

          setFieldsData({
            ...fields,
            ...fieldsToUpdate,
          });

          return;
        }

        break;

      default:
        break;
    }

    setFieldsData({
      ...fields,
      [name]: { ...fields[name], ...field },
    });
  };

  // const handleBlur = (e) => {
  // TODO: We need to develop a runtime blur handling with valiadation
  // };

  const checkConfirmation = () => {
    const { email, emailConfirm, password, passwordConfirm } = fields;
    const isEmailConfirmed = email.value === emailConfirm.value;
    const isPassConfirmed = password.value === passwordConfirm.value;
    const fieldsToUpdate = {};

    if (isEmailConfirmed && isPassConfirmed) {
      return true;
    }

    if (!isEmailConfirmed) {
      fieldsToUpdate[names.email] = {
        ...email,
        status: INPUT_STATUS.MISSMATCH,
      };
      fieldsToUpdate[names.emailConfirm] = {
        ...emailConfirm,
        status: INPUT_STATUS.MISSMATCH,
      };
    }

    if (!isPassConfirmed) {
      fieldsToUpdate[names.password] = {
        ...password,
        status: INPUT_STATUS.MISSMATCH,
      };
      fieldsToUpdate[names.passwordConfirm] = {
        ...passwordConfirm,
        status: INPUT_STATUS.MISSMATCH,
      };
    }

    setFieldsData({
      ...fields,
      ...fieldsToUpdate,
    });

    scrollUpTo(`#${names.email}`);

    return false;
  };

  const composeDetailsData = () => {
    const result = {};

    Object.keys(fields).forEach((key) => {
      const fieldValue = fields[key].value;

      result[key] =
        typeof fieldValue === 'string' ? fieldValue.trim() : fieldValue;
    });

    if (!result.password) {
      delete result.password;
      delete result.passwordConfirm;
    }

    delete result.emailConfirm;

    return result;
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (!checkConfirmation()) {
      return;
    }

    const validated = validate(fields);

    setFieldsData({
      ...fields,
      ...validated,
    });

    const invalidKey = Object.keys(validated)
      .filter((key) => validated[key].validationType !== VALIDATION_TYPES.NONE)
      .find((key) => validated[key].status === INPUT_STATUS.INVALID);

    if (!invalidKey) {
      actions.postDetails(composeDetailsData());
    } else {
      scrollUpTo(`#${names[invalidKey]}`);
    }
  };

  const renderStatus = (
    <div className={cn(s.status, { [s.msg]: updated, [s.error]: apiError })}>
      {updated && (
        <>
          <Done data-id="updated" className={s.icon} />
          <Typography color="green">Your details have been updated</Typography>
        </>
      )}
      {apiError && (
        <>
          <Exclamation data-id="error" className={s.icon} />
          <Typography color="red">
            Weâ€™re really sorry but an error has occurred saving your changes,
            please try again
          </Typography>
        </>
      )}
    </div>
  );

  const renderForm = isLoadingGet ? (
    <Spinner data-id="init-spinner" />
  ) : (
    <div className={s.formWrapper}>
      {(updated || apiError) && renderStatus}
      <form onSubmit={handleSubmit}>
        <SignIn names={names} fields={fields} handleChange={handleChange} />
        <Typography type="h4" className={s.heading}>
          Your profile information
        </Typography>
        <About names={names} fields={fields} handleChange={handleChange} />
        <Contact names={names} fields={fields} handleChange={handleChange} />
        <Address names={names} fields={fields} handleChange={handleChange} />
        <Additional names={names} fields={fields} handleChange={handleChange} />
        {isLoadingPost ? (
          <Spinner data-id="spinner" />
        ) : (
          <Button type="submit" className={s.submit}>
            Save
          </Button>
        )}
      </form>
    </div>
  );

  return (
    <section className={s.detailsWrapper}>
      <Typography type="h1" align="center" className={s.title}>
        Edit your personal details
      </Typography>
      <Typography type="h6" className={s.subtitle}>
        These details control how you sign-in to your buycar.co.uk account. Your
        email address is the address we will use to send any email
        correspondence to, whether we are responding to your question or sending
        you an update on your order.
      </Typography>
      {renderForm}
    </section>
  );
};

export const fieldShape = PropTypes.shape({
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  vaidationType: PropTypes.string,
  status: PropTypes.oneOf([
    INPUT_STATUS.VALID,
    INPUT_STATUS.INVALID,
    INPUT_STATUS.MISSMATCH,
    INPUT_STATUS.DEFAULT,
  ]),
});

export default Details;
