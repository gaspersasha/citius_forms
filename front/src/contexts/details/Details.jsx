import React, { createContext, useReducer, useContext } from 'react';
import initialState from './initial-state';
import { getDetails, postDetails, setDetailsState } from './actions';
import reducer from './reducer';
import middleware from './middleware';
import applyMiddlewares from '../apply-middlewares';

export const DetailsContext = createContext();

export const DetailsContextProvider = (props) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const enhancedDispatch = applyMiddlewares(middleware)(state, dispatch);
  const actions = {
    getDetails: () => enhancedDispatch(getDetails()),
    postDetails: (...args) => enhancedDispatch(postDetails(...args)),
    setDetailsState: (...args) => dispatch(setDetailsState(...args)),
  };

  return (
    <DetailsContext.Provider
      value={{
        state,
        actions,
      }}
      {...props}
    />
  );
};

export const useDetailsContext = () => useContext(DetailsContext);
