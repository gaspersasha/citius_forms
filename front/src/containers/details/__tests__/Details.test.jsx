import React from 'react';
import { INPUT_STATUS } from '~constants';
import { scrollUpTo } from '~utils';
import Details from '..';

jest.mock('~utils', () => ({
  ...jest.requireActual('../../../utils'),
  scrollUpTo: jest.fn(),
}));

const submitEvent = {
  preventDefault: jest.fn(),
  stopPropagation: jest.fn(),
};

describe('<Details />', () => {
  let wrapper;
  const mockGetDetails = jest.fn();
  const mockPostDetails = jest.fn();
  const mockSetDetailsState = jest.fn();

  const getWrapper = (method = shallow, state) => {
    global.mockUseDetailsContext({
      actions: {
        getDetails: mockGetDetails,
        postDetails: mockPostDetails,
        setDetailsState: mockSetDetailsState,
      },
      state: {
        data: global.mockData.details,
        isLoadingGet: false,
        isLoadingPost: false,
        updated: false,
        apiError: false,
        ...state,
      },
    });

    return method(<Details />);
  };

  it('renders without errors', () => {
    wrapper = getWrapper();

    expect(wrapper).toMatchSnapshot();
  });

  it('calls action to get data on mount', () => {
    wrapper = getWrapper(mount);

    expect(mockGetDetails).toBeCalled();
  });

  it('handles inital loading case', () => {
    wrapper = getWrapper(shallow, { isLoadingGet: true });

    expect(wrapper.find('[data-id="init-spinner"]').exists()).toBe(true);
  });

  it('handles submit loading case', () => {
    wrapper = getWrapper(shallow, { isLoadingPost: true });

    expect(wrapper.find('[data-id="spinner"]').exists()).toBe(true);
  });

  it('handles api error case', () => {
    wrapper = getWrapper(shallow, { apiError: true });

    expect(wrapper.find('[data-id="error"]').exists()).toBe(true);
  });

  it('handles successful update case', () => {
    wrapper = getWrapper(shallow, { updated: true });

    expect(wrapper.find('[data-id="updated"]').exists()).toBe(true);
  });

  it('handles fields change events', () => {
    wrapper = getWrapper();
    const getChild = () => wrapper.find('About');

    getChild().invoke('handleChange')({
      target: { name: 'validUKPassport', value: 'Yes' },
    });
    expect(getChild().prop('fields').validUKPassport.value).toEqual(true);

    getChild().invoke('handleChange')({
      target: { name: 'subscribed', checked: false },
    });
    expect(getChild().prop('fields').subscribed.value).toEqual(true);

    getChild().invoke('handleChange')({
      target: { name: 'dependants', value: '-1' },
    });
    expect(getChild().prop('fields').dependants.value).toEqual(
      global.mockData.details.dependants
    );
    getChild().invoke('handleChange')({
      target: { name: 'dependants', value: '2' },
    });
    expect(getChild().prop('fields').dependants.value).toEqual('2');

    getChild().invoke('handleChange')({
      target: { name: 'phone', value: '+' },
    });
    expect(getChild().prop('fields').phone.value).toEqual(
      global.mockData.details.phone
    );

    const value = {
      county: 'county',
      district: 'district',
      houseName: 'house',
      houseNumber: '1',
      street: 'street',
      town: 'town',
    };

    getChild().invoke('handleChange')({
      target: { value: 'RM82FJ', name: 'postcode' },
    });
    getChild().invoke('handleChange')({ target: { value, name: 'postcode' } });

    const fields = getChild().prop('fields');

    expect(fields.county.value).toEqual(value.county);
    expect(fields.district.value).toEqual(value.district);
    expect(fields.houseName.value).toEqual(value.houseName);
    expect(fields.houseNumber.value).toEqual(value.houseNumber);
    expect(fields.street.value).toEqual(value.street);
    expect(fields.town.value).toEqual(value.town);
  });

  it('calls action to set the default apiError and updated values on change', () => {
    wrapper = getWrapper(shallow, { updated: true });

    wrapper.find('About').invoke('handleChange')({
      target: { name: 'phone', value: '' },
    });
    expect(mockSetDetailsState).toBeCalledWith({
      apiError: false,
      updated: false,
    });
  });

  it('handles submit valid case', () => {
    wrapper = getWrapper();

    wrapper.find('form').simulate('submit', submitEvent);
    expect(mockPostDetails).toBeCalledWith(global.mockData.details);
  });

  it('handles submit invalid case', () => {
    wrapper = getWrapper();

    wrapper.find('About').invoke('handleChange')({
      target: { name: 'phone', value: '0' },
    });
    wrapper.find('form').simulate('submit', submitEvent);

    expect(scrollUpTo).toBeCalledWith('#phone');
  });

  it('handles submit no confirmation case', () => {
    wrapper = getWrapper();
    const getChild = () => wrapper.find('About');

    wrapper.find('About').invoke('handleChange')({
      target: { name: 'emailConfirm', value: INPUT_STATUS.INVALID },
    });

    wrapper.find('About').invoke('handleChange')({
      target: { name: 'password', value: INPUT_STATUS.INVALID },
    });

    wrapper.find('form').simulate('submit', submitEvent);

    const fields = getChild().prop('fields');

    expect(fields.email.status).toEqual(INPUT_STATUS.MISSMATCH);
    expect(fields.emailConfirm.status).toEqual(INPUT_STATUS.MISSMATCH);
    expect(fields.password.status).toEqual(INPUT_STATUS.MISSMATCH);
    expect(fields.passwordConfirm.status).toEqual(INPUT_STATUS.MISSMATCH);
    expect(scrollUpTo).toBeCalledWith('#email');
  });
});
