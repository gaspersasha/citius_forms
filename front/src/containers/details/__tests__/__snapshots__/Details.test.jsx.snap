// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> renders without errors 1`] = `
<section
  className="detailsWrapper"
>
  <Typography
    align="center"
    className="title"
    color="dark"
    type="h1"
  >
    Edit your personal details
  </Typography>
  <Typography
    align="left"
    className="subtitle"
    color="dark"
    type="h6"
  >
    These details control how you sign-in to your buycar.co.uk account. Your email address is the address we will use to send any email correspondence to, whether we are responding to your question or sending you an update on your order.
  </Typography>
  <div
    className="formWrapper"
  >
    <form
      onSubmit={[Function]}
    >
      <SignIn
        fields={
          Object {
            "alternativePhone": Object {
              "status": "default",
              "validationType": "altPhone",
              "value": "02057685678",
            },
            "company": Object {
              "status": "default",
              "validationType": "none",
              "value": "company",
            },
            "county": Object {
              "status": "default",
              "validationType": "none",
              "value": "county",
            },
            "dateOfBirth": Object {
              "status": "default",
              "validationType": "date",
              "value": "11/11/2000",
            },
            "dependants": Object {
              "status": "default",
              "validationType": "none",
              "value": "1",
            },
            "district": Object {
              "status": "default",
              "validationType": "none",
              "value": "district",
            },
            "drivingLicenceType": Object {
              "status": "default",
              "validationType": "none",
              "value": "FULL",
            },
            "email": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "emailConfirm": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "firstName": Object {
              "status": "default",
              "validationType": "name",
              "value": "first",
            },
            "houseName": Object {
              "status": "default",
              "validationType": "none",
              "value": "housename",
            },
            "houseNumber": Object {
              "status": "default",
              "validationType": "none",
              "value": "100",
            },
            "lastName": Object {
              "status": "default",
              "validationType": "name",
              "value": "last",
            },
            "maritalStatus": Object {
              "status": "default",
              "validationType": "none",
              "value": "SINGLE",
            },
            "middleName": Object {
              "status": "default",
              "validationType": "notRequiredText",
              "value": "mid",
            },
            "password": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "passwordConfirm": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "phone": Object {
              "status": "default",
              "validationType": "phone",
              "value": "02057685678",
            },
            "postcode": Object {
              "status": "default",
              "validationType": "postCode",
              "value": "E11 1JD",
            },
            "street": Object {
              "status": "default",
              "validationType": "none",
              "value": "street",
            },
            "subscribed": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
            "title": Object {
              "status": "default",
              "validationType": "notEmpty",
              "value": "MR",
            },
            "town": Object {
              "status": "default",
              "validationType": "none",
              "value": "town",
            },
            "validUKPassport": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
          }
        }
        handleChange={[Function]}
        names={
          Object {
            "alternativePhone": "alternativePhone",
            "company": "company",
            "county": "county",
            "dateOfBirth": "dateOfBirth",
            "dependants": "dependants",
            "district": "district",
            "drivingLicenceType": "drivingLicenceType",
            "email": "email",
            "emailConfirm": "emailConfirm",
            "firstName": "firstName",
            "houseName": "houseName",
            "houseNumber": "houseNumber",
            "lastName": "lastName",
            "maritalStatus": "maritalStatus",
            "middleName": "middleName",
            "password": "password",
            "passwordConfirm": "passwordConfirm",
            "phone": "phone",
            "postcode": "postcode",
            "street": "street",
            "subscribed": "subscribed",
            "title": "title",
            "town": "town",
            "validUKPassport": "validUKPassport",
          }
        }
      />
      <Typography
        align="left"
        className="heading"
        color="dark"
        type="h4"
      >
        Your profile information
      </Typography>
      <About
        fields={
          Object {
            "alternativePhone": Object {
              "status": "default",
              "validationType": "altPhone",
              "value": "02057685678",
            },
            "company": Object {
              "status": "default",
              "validationType": "none",
              "value": "company",
            },
            "county": Object {
              "status": "default",
              "validationType": "none",
              "value": "county",
            },
            "dateOfBirth": Object {
              "status": "default",
              "validationType": "date",
              "value": "11/11/2000",
            },
            "dependants": Object {
              "status": "default",
              "validationType": "none",
              "value": "1",
            },
            "district": Object {
              "status": "default",
              "validationType": "none",
              "value": "district",
            },
            "drivingLicenceType": Object {
              "status": "default",
              "validationType": "none",
              "value": "FULL",
            },
            "email": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "emailConfirm": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "firstName": Object {
              "status": "default",
              "validationType": "name",
              "value": "first",
            },
            "houseName": Object {
              "status": "default",
              "validationType": "none",
              "value": "housename",
            },
            "houseNumber": Object {
              "status": "default",
              "validationType": "none",
              "value": "100",
            },
            "lastName": Object {
              "status": "default",
              "validationType": "name",
              "value": "last",
            },
            "maritalStatus": Object {
              "status": "default",
              "validationType": "none",
              "value": "SINGLE",
            },
            "middleName": Object {
              "status": "default",
              "validationType": "notRequiredText",
              "value": "mid",
            },
            "password": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "passwordConfirm": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "phone": Object {
              "status": "default",
              "validationType": "phone",
              "value": "02057685678",
            },
            "postcode": Object {
              "status": "default",
              "validationType": "postCode",
              "value": "E11 1JD",
            },
            "street": Object {
              "status": "default",
              "validationType": "none",
              "value": "street",
            },
            "subscribed": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
            "title": Object {
              "status": "default",
              "validationType": "notEmpty",
              "value": "MR",
            },
            "town": Object {
              "status": "default",
              "validationType": "none",
              "value": "town",
            },
            "validUKPassport": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
          }
        }
        handleChange={[Function]}
        names={
          Object {
            "alternativePhone": "alternativePhone",
            "company": "company",
            "county": "county",
            "dateOfBirth": "dateOfBirth",
            "dependants": "dependants",
            "district": "district",
            "drivingLicenceType": "drivingLicenceType",
            "email": "email",
            "emailConfirm": "emailConfirm",
            "firstName": "firstName",
            "houseName": "houseName",
            "houseNumber": "houseNumber",
            "lastName": "lastName",
            "maritalStatus": "maritalStatus",
            "middleName": "middleName",
            "password": "password",
            "passwordConfirm": "passwordConfirm",
            "phone": "phone",
            "postcode": "postcode",
            "street": "street",
            "subscribed": "subscribed",
            "title": "title",
            "town": "town",
            "validUKPassport": "validUKPassport",
          }
        }
      />
      <Contact
        fields={
          Object {
            "alternativePhone": Object {
              "status": "default",
              "validationType": "altPhone",
              "value": "02057685678",
            },
            "company": Object {
              "status": "default",
              "validationType": "none",
              "value": "company",
            },
            "county": Object {
              "status": "default",
              "validationType": "none",
              "value": "county",
            },
            "dateOfBirth": Object {
              "status": "default",
              "validationType": "date",
              "value": "11/11/2000",
            },
            "dependants": Object {
              "status": "default",
              "validationType": "none",
              "value": "1",
            },
            "district": Object {
              "status": "default",
              "validationType": "none",
              "value": "district",
            },
            "drivingLicenceType": Object {
              "status": "default",
              "validationType": "none",
              "value": "FULL",
            },
            "email": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "emailConfirm": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "firstName": Object {
              "status": "default",
              "validationType": "name",
              "value": "first",
            },
            "houseName": Object {
              "status": "default",
              "validationType": "none",
              "value": "housename",
            },
            "houseNumber": Object {
              "status": "default",
              "validationType": "none",
              "value": "100",
            },
            "lastName": Object {
              "status": "default",
              "validationType": "name",
              "value": "last",
            },
            "maritalStatus": Object {
              "status": "default",
              "validationType": "none",
              "value": "SINGLE",
            },
            "middleName": Object {
              "status": "default",
              "validationType": "notRequiredText",
              "value": "mid",
            },
            "password": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "passwordConfirm": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "phone": Object {
              "status": "default",
              "validationType": "phone",
              "value": "02057685678",
            },
            "postcode": Object {
              "status": "default",
              "validationType": "postCode",
              "value": "E11 1JD",
            },
            "street": Object {
              "status": "default",
              "validationType": "none",
              "value": "street",
            },
            "subscribed": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
            "title": Object {
              "status": "default",
              "validationType": "notEmpty",
              "value": "MR",
            },
            "town": Object {
              "status": "default",
              "validationType": "none",
              "value": "town",
            },
            "validUKPassport": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
          }
        }
        handleChange={[Function]}
        names={
          Object {
            "alternativePhone": "alternativePhone",
            "company": "company",
            "county": "county",
            "dateOfBirth": "dateOfBirth",
            "dependants": "dependants",
            "district": "district",
            "drivingLicenceType": "drivingLicenceType",
            "email": "email",
            "emailConfirm": "emailConfirm",
            "firstName": "firstName",
            "houseName": "houseName",
            "houseNumber": "houseNumber",
            "lastName": "lastName",
            "maritalStatus": "maritalStatus",
            "middleName": "middleName",
            "password": "password",
            "passwordConfirm": "passwordConfirm",
            "phone": "phone",
            "postcode": "postcode",
            "street": "street",
            "subscribed": "subscribed",
            "title": "title",
            "town": "town",
            "validUKPassport": "validUKPassport",
          }
        }
      />
      <Address
        fields={
          Object {
            "alternativePhone": Object {
              "status": "default",
              "validationType": "altPhone",
              "value": "02057685678",
            },
            "company": Object {
              "status": "default",
              "validationType": "none",
              "value": "company",
            },
            "county": Object {
              "status": "default",
              "validationType": "none",
              "value": "county",
            },
            "dateOfBirth": Object {
              "status": "default",
              "validationType": "date",
              "value": "11/11/2000",
            },
            "dependants": Object {
              "status": "default",
              "validationType": "none",
              "value": "1",
            },
            "district": Object {
              "status": "default",
              "validationType": "none",
              "value": "district",
            },
            "drivingLicenceType": Object {
              "status": "default",
              "validationType": "none",
              "value": "FULL",
            },
            "email": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "emailConfirm": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "firstName": Object {
              "status": "default",
              "validationType": "name",
              "value": "first",
            },
            "houseName": Object {
              "status": "default",
              "validationType": "none",
              "value": "housename",
            },
            "houseNumber": Object {
              "status": "default",
              "validationType": "none",
              "value": "100",
            },
            "lastName": Object {
              "status": "default",
              "validationType": "name",
              "value": "last",
            },
            "maritalStatus": Object {
              "status": "default",
              "validationType": "none",
              "value": "SINGLE",
            },
            "middleName": Object {
              "status": "default",
              "validationType": "notRequiredText",
              "value": "mid",
            },
            "password": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "passwordConfirm": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "phone": Object {
              "status": "default",
              "validationType": "phone",
              "value": "02057685678",
            },
            "postcode": Object {
              "status": "default",
              "validationType": "postCode",
              "value": "E11 1JD",
            },
            "street": Object {
              "status": "default",
              "validationType": "none",
              "value": "street",
            },
            "subscribed": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
            "title": Object {
              "status": "default",
              "validationType": "notEmpty",
              "value": "MR",
            },
            "town": Object {
              "status": "default",
              "validationType": "none",
              "value": "town",
            },
            "validUKPassport": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
          }
        }
        handleChange={[Function]}
        names={
          Object {
            "alternativePhone": "alternativePhone",
            "company": "company",
            "county": "county",
            "dateOfBirth": "dateOfBirth",
            "dependants": "dependants",
            "district": "district",
            "drivingLicenceType": "drivingLicenceType",
            "email": "email",
            "emailConfirm": "emailConfirm",
            "firstName": "firstName",
            "houseName": "houseName",
            "houseNumber": "houseNumber",
            "lastName": "lastName",
            "maritalStatus": "maritalStatus",
            "middleName": "middleName",
            "password": "password",
            "passwordConfirm": "passwordConfirm",
            "phone": "phone",
            "postcode": "postcode",
            "street": "street",
            "subscribed": "subscribed",
            "title": "title",
            "town": "town",
            "validUKPassport": "validUKPassport",
          }
        }
      />
      <Additional
        fields={
          Object {
            "alternativePhone": Object {
              "status": "default",
              "validationType": "altPhone",
              "value": "02057685678",
            },
            "company": Object {
              "status": "default",
              "validationType": "none",
              "value": "company",
            },
            "county": Object {
              "status": "default",
              "validationType": "none",
              "value": "county",
            },
            "dateOfBirth": Object {
              "status": "default",
              "validationType": "date",
              "value": "11/11/2000",
            },
            "dependants": Object {
              "status": "default",
              "validationType": "none",
              "value": "1",
            },
            "district": Object {
              "status": "default",
              "validationType": "none",
              "value": "district",
            },
            "drivingLicenceType": Object {
              "status": "default",
              "validationType": "none",
              "value": "FULL",
            },
            "email": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "emailConfirm": Object {
              "status": "default",
              "validationType": "email",
              "value": "email@test.test",
            },
            "firstName": Object {
              "status": "default",
              "validationType": "name",
              "value": "first",
            },
            "houseName": Object {
              "status": "default",
              "validationType": "none",
              "value": "housename",
            },
            "houseNumber": Object {
              "status": "default",
              "validationType": "none",
              "value": "100",
            },
            "lastName": Object {
              "status": "default",
              "validationType": "name",
              "value": "last",
            },
            "maritalStatus": Object {
              "status": "default",
              "validationType": "none",
              "value": "SINGLE",
            },
            "middleName": Object {
              "status": "default",
              "validationType": "notRequiredText",
              "value": "mid",
            },
            "password": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "passwordConfirm": Object {
              "status": "default",
              "validationType": "password",
              "value": "",
            },
            "phone": Object {
              "status": "default",
              "validationType": "phone",
              "value": "02057685678",
            },
            "postcode": Object {
              "status": "default",
              "validationType": "postCode",
              "value": "E11 1JD",
            },
            "street": Object {
              "status": "default",
              "validationType": "none",
              "value": "street",
            },
            "subscribed": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
            "title": Object {
              "status": "default",
              "validationType": "notEmpty",
              "value": "MR",
            },
            "town": Object {
              "status": "default",
              "validationType": "none",
              "value": "town",
            },
            "validUKPassport": Object {
              "status": "default",
              "validationType": "none",
              "value": false,
            },
          }
        }
        handleChange={[Function]}
        names={
          Object {
            "alternativePhone": "alternativePhone",
            "company": "company",
            "county": "county",
            "dateOfBirth": "dateOfBirth",
            "dependants": "dependants",
            "district": "district",
            "drivingLicenceType": "drivingLicenceType",
            "email": "email",
            "emailConfirm": "emailConfirm",
            "firstName": "firstName",
            "houseName": "houseName",
            "houseNumber": "houseNumber",
            "lastName": "lastName",
            "maritalStatus": "maritalStatus",
            "middleName": "middleName",
            "password": "password",
            "passwordConfirm": "passwordConfirm",
            "phone": "phone",
            "postcode": "postcode",
            "street": "street",
            "subscribed": "subscribed",
            "title": "title",
            "town": "town",
            "validUKPassport": "validUKPassport",
          }
        }
      />
      <Button
        className="submit"
        disabled={false}
        href=""
        onClick={null}
        selected={false}
        styleType="primary"
        type="submit"
      >
        Save
      </Button>
    </form>
  </div>
</section>
`;
